tensorflow/contrib/cmake/CMakeLists.txt

```diff
@@ -153,13 +156,25 @@ endif()
 
 # MSVC SIMD instructions
 if (tensorflow_WIN_CPU_SIMD_OPTIONS)
+  include(CheckCXXCompilerFlag)
+  add_definitions(-DINTEL_MKL -DEIGEN_USE_VML)
+  if (NOT tensorflow_ENABLE_MKLDNN_SUPPORT)
+    add_definitions(-DINTEL_MKL_ML)
+  endif()
+  CHECK_CXX_COMPILER_FLAG("-fopenmp" COMPILER_OPT_OPENMP_SUPPORT)
+  if (COMPILER_OPT_OPENMP_SUPPORT)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp")
+  endif()
   if (WIN32)
-    CHECK_CXX_COMPILER_FLAG("${tensorflow_WIN_CPU_SIMD_OPTIONS}" COMPILER_OPT_WIN_CPU_SIMD_SUPPORTED)
-    if(COMPILER_OPT_WIN_CPU_SIMD_SUPPORTED)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${tensorflow_WIN_CPU_SIMD_OPTIONS}")
-    else()
-      message(FATAL_ERROR "${tensorflow_WIN_CPU_SIMD_OPTIONS} not supported")
-    endif()
+    set(WIN_SIMD_FLAGS "/arch:AVX2" "/arch:AVX" "/arch:SSE2" "/arch:SSE")
+    foreach(simd_flag ${WIN_SIMD_FLAGS})
+      CHECK_CXX_COMPILER_FLAG(${simd_flag} COMPILER_OPT_WIN_CPU_SIMD_SUPPORTED)
+      if(COMPILER_OPT_WIN_CPU_SIMD_SUPPORTED)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${simd_flag}")
+        message(STATUS "Using ${simd_flag} as SIMD flag")
+        break() # Use the former as the priority
```

Comment:
Does this loop over `WIN_SIMD_FLAGS` in order? In that case, would it make more sense to start with `/arch:SSE` and end with `/arch:AVX2`?

Reply:
Cross-compilation is problematic at this time because some of the compilation products of tensorflow are executables that need to be executed on the compilation machine. Setting the AVX2 flag on the build machine that CPU does not support AVX2 will cause the compilation to fail.

Reply:
@mrry Thanks.
I revert this loop, and keep using `tensorflow_WIN_CPU_SIMD_OPTIONS` to specify SIMD flags